### PR TITLE
components: Add options to customize FreeRTOS behaviour.

### PR DESCRIPTION
**Pull request by danicampora**
These options are:

    - SUPPORT_STATIC_ALLOCATION
    - ENABLE_STATIC_TASK_CLEAN_UP_HOOK
    - TIMER_TASK_PRIORITY
    - TIMER_TASK_STACK_DEPTH
    - TIMER_QUEUE_LENGTH

### CODE REVIEW

File: components/freertos/Kconfig [hunk 1]
```diff
@@ -197,6 +197,81 @@ config FREERTOS_MAX_TASK_NAME_LEN
...
+        (although FreeRTOS includes allocation schemes that can overcome most objections).
+
+config ENABLE_STATIC_TASK_CLEAN_UP_HOOK
+    bool "Enable static task clean up hook"
+    depends on SUPPORT_STATIC_ALLOCATION
+    default n
+    help
+        Enable this option to make FreeRTOS call the static task clean up hook when a task is deleted.
+
+        Bare in mind that if this option is enabled you will need to implement the following function:
```
Comment (projectgus) [inline components/freertos/Kconfig]: Typo: bear

**Comment by projectgus** [comment]
This looks good to me. There's one typo, and I'd like to reduce the minimum configurable stack size so that it can be the set the same as currently, but I can make these changes when I cherry-pick the commit.

Thanks! :)

**Comment by danicampora** [comment]
@projectgus thanks :-)

**Comment by projectgus** [comment]
Merged in 01ad387. Thanks again!

**Pull request closed without merging**